terraform/jenkins/main.tf

```diff
@@ -50,7 +50,8 @@ resource "aws_instance" "jenkins_server" {
 
 resource "aws_ebs_volume" "jenkins_home" {
   availability_zone = var.vpc_azs[0]
-  size              = var.jenkins_home_size
+  size              = var.snapshot_id == null ? var.jenkins_home_size : null
```

Comment:
@sergii-msn 

In case when **snapshot_id** is defined there will be created a new volume with undefined size.
Are you sure the size for a volume can be **null** ?

Reply:
yes, new volume will be created considering volumesize based on snapshot size (500gb)